### PR TITLE
Setup improvements

### LINKED ISSUE #11: Setup improvements
**Enhancements to initial setup**

> A dependency on eslint and a sensible setup. I prefer eslint-config-react-app though it is annoying to setup because of peer dependencies. This will bring this project into line with the linting used in (I think) every other React project at neontribe.
> A setup with husky and lint-staged to lint and prettify on commit.
> All dependencies to be pinned unmasked (e.g. "gatsby": "2.0.53" instead of "gatsby": "^2.0.53" )

**Plan**
- [x] Install `eslint-config-react-app` and required extras
- [x] Setup husky and lint-staged, and test by committing a poorly formatted file
- [x] Remove "compatible with" in dependency versions

### PR DESCRIPTION
**Pull request by Nikomus**
Fixes #11 

## Description

- Adds eslint and husky

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,13 +21,24 @@
   ],
   "license": "MIT",
   "scripts": {
+    "precommit": "lint-staged",
+    "postcommit": "git update-index --again",
```
Comment (marxian) [inline package.json]: This is exciting. What does it achieve?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+node_modules
+build
+coverage
```
Comment (marxian) [inline .eslintignore]: I've a hunch we should ignore `public` too - since that is gatsby's build output.

File: package.json [hunk 3]
```diff
@@ -36,5 +47,20 @@
...
     "url": "https://github.com/gatsbyjs/gatsby-starter-default"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,json,css,md}": [
+      "prettier --single-quote --trailing-comma es5 --write",
```
Comment (marxian) [inline package.json]: I think this duplicates settings in `.prettierrc`?

File: package.json (hunk 1 above)
Comment (Nikomus) [reply package.json]: @marxian We've found this to help with editors with built-in git clients from reverting linting.

Comment (marxian) [reply package.json]: :+1: I get it. That's useful knowledge. Thanks.

Comment (Nikomus) [reply package.json]: From here, btw: https://www.npmjs.com/package/lint-staged#using-with-jetbrains-ides-webstorm-pycharm-intellij-idea-rubymine-etc

File: .eslintignore (hunk 2 above)
Comment (Nikomus) [reply .eslintignore]: Seems to be useful to add `static` and `.cache` too

File: package.json [hunk 4]
```diff
@@ -21,13 +21,24 @@
...
     "develop": "gatsby develop",
     "start": "yarn run develop",
     "format": "prettier --write \"src/**/*.js\"",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "devDependencies": {
+    "babel-eslint": "9.x",
+    "eslint": "5.x",
+    "eslint-config-react-app": "^3.0.5",
+    "eslint-plugin-flowtype": "2.x",
```
Comment (katjam) [inline package.json]: Do we need flowtype? I'd be happy not to use flow with this project. I found it hindered more than helped in MOMO.

Comment (marxian) [reply package.json]: It is a required peerDependency of eslint-config-react-app

**Pull request merged**